picard/formats/id3.py

```diff
@@ -869,6 +691,304 @@ def _parse_lrc_text(self, text):
+        self._save_number_tag(tags, metadata, 'movementnumber', 'movementtotal', 'MVIN')
+
+    def _save_number_tag(self, tags, metadata, number_tag, total_tag, frame_id):
+        """Generic method to save a number tag with optional total."""
+        if number_tag in metadata:
+            if total_tag in metadata:
+                text = '%s/%s' % (metadata[number_tag], metadata[total_tag])
+            else:
+                text = metadata[number_tag]
+            tags.add(getattr(id3, frame_id)(
+                encoding=Id3Encoding.LATIN1,
+                text=id3text(text, Id3Encoding.LATIN1)
+            ))
+
+    def _save_images(self, tags, metadata):
+        """Save cover art images to tags."""
+        images_to_save = list(metadata.images.to_be_saved_to_tags())
+        if not images_to_save:
+            return
+
+        tags.delall('APIC')
+
+        counters = Counter()
+        for image in images_to_save:
+            desc = desctag = image.comment
+            if counters[desc] > 0:
+                if desc:
+                    desctag = "%s (%i)" % (desc, counters[desc])
+                else:
+                    desctag = "(%i)" % counters[desc]
+            counters[desc] += 1
+            tags.add(id3.APIC(
+                encoding=Id3Encoding.LATIN1,
+                mime=image.mimetype,
+                type=image.id3_type,
+                desc=id3text(desctag, Id3Encoding.LATIN1),
+                data=image.data
+            ))
+
+    def _save_comment_tag(self, tags, name, values, encoding):
+        """Save comment tag to ID3 frames."""
+        (lang, desc) = parse_comment_tag(name)
+        if desc.lower()[:4] == 'itun':
+            tags.delall('COMM:' + desc)
+            tags.add(id3.COMM(encoding=Id3Encoding.LATIN1, desc=desc, lang='eng', text=[v + '\x00' for v in values]))
+        else:
+            tags.add(id3.COMM(encoding=encoding, desc=desc, lang=lang, text=values))
+
+    def _save_lyrics_tag(self, tags, name, values, encoding):
+        """Save lyrics tag to ID3 frames."""
+        if ':' in name:
+            desc = name.split(':', 1)[1]
+        else:
+            desc = ''
+        for value in values:
+            tags.add(id3.USLT(encoding=encoding, desc=desc, text=value))
+
+    def _save_synced_lyrics_tag(self, tags, name, values, encoding):
+        """Save synchronized lyrics tag to ID3 frames."""
+        (lang, desc) = parse_subtag(name)
+        for value in values:
+            sylt_lyrics = self._parse_lrc_text(value)
+            # If the text does not contain any timestamps, the tag is not added
+            if sylt_lyrics:
+                tags.add(id3.SYLT(encoding=encoding, lang=lang, format=2, type=1, desc=desc, text=sylt_lyrics))
+
+    def _save_musicbrainz_recording_id(self, tags, values):
+        """Save MusicBrainz recording ID to UFID frame."""
+        tags.add(id3.UFID(owner="http://musicbrainz.org", data=bytes(values[0], 'ascii')))
```

Comment:
I'm wondering if it makes sense to store the "http://musicbrainz.org" as a constant and serve it from there?  I'm pretty sure it (or the https:// version) is used elsewhere in the code.  If so, it may be worthy of a separate PR because it isn't directly related to what you're doing here.

Reply:
Done in https://github.com/metabrainz/picard/commit/1d95d9f087b923907b0012e55fd1c6a47d027423